src/components/Main/MainCarousel/index.jsx

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Box, IconButton } from '@mui/material';
+import { ArrowBackIosNew, ArrowForwardIos } from '@mui/icons-material';
+import CardSlide from './CardSlide';
+import { styled } from '@mui/system';
+
+const MainCarousel = ({ waitingOtts, slideGap }) => {
+  const slideRef = useRef(null);
+  const totalSlide = waitingOtts.length - 1;
+  const [activeSlide, setActiveSlide] = useState(0);
+
+  useEffect(() => {
+    slideRef.current.style.transition = 'all 0.5s ease-in-out';
+    slideRef.current.style.transform = `translateX(calc(-${
+      activeSlide * 100
+    }% - ${activeSlide * slideGap}px))`;
+  }, [activeSlide, slideGap]);
+
+  const handleNextSlide = e => {
+    setActiveSlide(prevSlide => prevSlide + 1);
+    if (activeSlide === totalSlide) {
+      setActiveSlide(0);
+    }
+  };
```

Comment:
```suggestion
  const handleNextSlide = () => {
    setActiveSlide(prevSlide => prevSlide === totalSlide ? 0 : prevSlide + 1);
  };
```
요렇게 작성하면 좀 더 코드를 줄일 수 있을 것 같아요! 
그리고 e는 사용하지 않아서 제외해도 좋을 것 같습니다.

Reply:
간결하고 명확하고 너무 좋네요